thorin/pass/pass.cpp

```diff
@@ -46,14 +46,14 @@ void PassMan::run() {
     for (auto&& pass : passes_) world().ILOG(" + {}", pass->name());
     world().debug_dump();
 
+    for (auto&& pass : passes_) pass->prepare();
```

Comment:
what is the reason to move the prepare before the `analyzed` and `curr_state` initialization?
The main difference I see is if the preparation introduces new externals.

Reply:
yeah exactly, that's been the motivation here.
If externals need to be changed, this PR does so by creating a new nom and marking the old one as internal and the new one as external...